### PR TITLE
Support dynamic resource profiles for tracing and metrics

### PR DESCRIPTION
**Pull request by chrkl**
**Description**

The resource requirements of the OpenTelemetry Collector depend on the number of existing pipelines. To ensure a minimum throughput for TracePipelines and MetricPipelines, this PR adjusts the CPU and memory limit of the collector deployment based on the number of pipelines.

Changes proposed in this pull request:

- Adjust otel collector resources based on the number of pipelines
- Deploy 2 collector instances for better availability
- Add session affinity to the OTLP services to support stateful processors in future
- Disable Istio sidecar injection for the telemetry-operator

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/17377

### CODE REVIEW

File: test/e2e/metrics_test.go [hunk 1]
```diff
@@ -50,13 +52,21 @@ var _ = Describe("Metrics", func() {
...
 			Eventually(func(g Gomega) {
-				key := types.NamespacedName{Name: metricGatewayBaseName, Namespace: kymaSystemNamespaceName}
-				ready, err := verifiers.IsDeploymentReady(ctx, k8sClient, key)
+				ready, err := verifiers.IsDeploymentReady(ctx, k8sClient, metricGatewayName)
 				g.Expect(err).ShouldNot(HaveOccurred())
 				g.Expect(ready).To(BeTrue())
 			}, timeout, interval).Should(Succeed())
 		})
 
+		It("Should have 2 trace collector replicas", func() {
```
Comment (dennis-ge) [inline test/e2e/metrics_test.go:61]: this test description is wrong

File: test/e2e/tracing_test.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ var _ = Describe("Tracing", func() {
 		It("Should have a running trace collector deployment", func() {
 			Eventually(func(g Gomega) {
 				key := types.NamespacedName{Name: traceCollectorBaseName, Namespace: kymaSystemNamespaceName}
-				ready, err := verifiers.IsDeploymentReady(ctx, k8sClient, key)
+				ready, err := verifiers.IsDeploymentReady(context.Background(), k8sClient, key)
 				g.Expect(err).NotTo(HaveOccurred())
 				g.Expect(ready).To(BeTrue())
 			}, timeout, interval).Should(Succeed())
```
Comment (dennis-ge) [inline test/e2e/tracing_test.go:58]: The test case checking for two trace collector replicas is missing

**Review by dennis-ge — CHANGES_REQUESTED** [review]
Looks good. Found some minor inconsistencies in the e2e test

File: test/e2e/tracing_test.go (hunk 2 above)
Comment (chrkl) [reply test/e2e/tracing_test.go:59]: Good catch, done

**Review by dennis-ge — APPROVED** [review]

**Review by dennis-ge — APPROVED** [review]

**Review by dennis-ge — APPROVED** [review]

**Pull request merged**